subsys/bluetooth/host/conn.c

```diff
@@ -861,7 +862,7 @@ static bool dont_have_viewbufs(void)
 #endif	/* CONFIG_BT_CONN_TX */
 }
 
-static bool dont_have_methods(struct bt_conn *conn)
+__maybe_unused static bool dont_have_methods(struct bt_conn *conn)
```

Comment:
Is `__maybe_unused` necessary? We have many `static` functions that are guarded by e.g. `IS_ENABLED` that don't require this. 

Maybe `CHECKIF` should use `IS_ENABLED` rather than `#if`.

Reply:
An alternative is to also use a regular `if` for guarding this call. We don't really have any guidelines on when to use one over the other, but my rule of thumb is that I use `CHECKIF` for values that are always incorrect and/or checkable by the caller (such as values out of range, etc.), but not for state-related checks. 

It's OK to keep as is though, assuming that coverity likes `__maybe_unused`

Reply:
I use CHECKIF so the people that _have_ to go fast can skip the user-data checkloop.